File: backtest-core/src/main/java/com/yuanzhixiang/bt/domain/model/valobj/Price.java
package com.yuanzhixiang.bt.domain.model.valobj;

import java.math.BigDecimal;

import com.yuanzhixiang.bt.engine.Local.LocalMap;
import com.yuanzhixiang.bt.engine.LocalVariable;

/**
 * @author yuanzhixiang
 */
public class Price implements LocalVariable {

    private final Double price;

    private final BigDecimal adjustment;

    public Price(Double price, BigDecimal adjustment) {
        this.price = price;
        this.adjustment = adjustment;
    }

    public Double getPrice() {
        return price;
    }

    public BigDecimal getAdjustment() {
        return adjustment;
    }

    private LocalMap localMap;

    @Override
    public LocalMap getLocalMap() {
        return localMap;
    }

    @Override
    public void setLocalMap(LocalMap localMap) {
        this.localMap = localMap;
    }
}

File: backtest-core/src/main/java/com/yuanzhixiang/bt/engine/domain/Price.kt
package com.yuanzhixiang.bt.engine.domain

import com.yuanzhixiang.bt.engine.Local.LocalMap
import com.yuanzhixiang.bt.engine.LocalVariable
import java.math.BigDecimal

/**
 * @author Yuan Zhixiang
 */
class Price(val price: Double, val adjustment: BigDecimal) : LocalVariable {
    private var localMap: LocalMap? = null
    override fun getLocalMap(): LocalMap {
        return localMap!!
    }

    override fun setLocalMap(localMap: LocalMap) {
        this.localMap = localMap
    }
}